tests/graph/testObjectBoostGraphInterface.cpp

```diff
@@ -0,0 +1,625 @@
+ *
+ * This file is part of the DGtal library.
+ */
+
+///////////////////////////////////////////////////////////////////////////////
+#include "DGtalCatch.h"
+#include <iostream>
+#include <queue>
+#include <boost/property_map/property_map.hpp>
+#include <boost/pending/queue.hpp>
+#include "DGtal/base/Common.h"
+#include "DGtal/graph/ObjectBoostGraphInterface.h"
+#include "DGtal/helpers/StdDefs.h"
+#include "DGtal/topology/Object.h"
+#include "DGtal/topology/DigitalTopology.h"
+#include "DGtal/topology/SurfelAdjacency.h"
+/// JOL: Since using boost::vertices is enforced before we define it,
+/// the compiler is unable to find our function boost::vertices. We
+/// *must* include graph_concepts.hpp after defining our graph wrapper.
+#include <boost/graph/graph_concepts.hpp>
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/copy.hpp>
+#include <boost/graph/breadth_first_search.hpp>
+#include <boost/graph/connected_components.hpp>
+#include <boost/graph/stoer_wagner_min_cut.hpp>
+#include <boost/graph/boykov_kolmogorov_max_flow.hpp>
+#include <boost/graph/filtered_graph.hpp>
+///////////////////////////////////////////////////////////////////////////
+using namespace std;
+using namespace DGtal;
+
+///////////////////////////////////////////////////////////////////////////
+// Fixture for object from a diamond set and DT26_6 topology.
+///////////////////////////////////////////////////////////////////////////
+struct Fixture_object_diamond_with_hole {
+  ///////////////////////////////////////////////////////////
+  // type aliases
+  ///////////////////////////////////////////////////////////
+  using Point  = DGtal::Z3i::Point;
+  using Domain = DGtal::Z3i::Domain;
+  using KSpace = DGtal::Z3i::KSpace;
+
+  using FixtureSurfelAdjacency =
+    DGtal::SurfelAdjacency<DGtal::Z3i::KSpace::dimension> ;
+  using FixtureDigitalTopology =
+    DGtal::Z3i::DT26_6;
+  using FixtureDigitalSet =
+    DGtal::DigitalSetByAssociativeContainer<DGtal::Z3i::Domain , std::unordered_set< typename DGtal::Z3i::Domain::Point> >;
+  using FixtureObject =
+    DGtal::Object<FixtureDigitalTopology, FixtureDigitalSet>;
+
+  ///////////////////////////////////////////////////////////
+  // fixture data
+  FixtureObject obj_fixture;
+  ///////////////////////////////////////////////////////////
+
+  ///////////////////////////////////////////////////////////
+  // Constructor
+  ///////////////////////////////////////////////////////////
+  Fixture_object_diamond_with_hole() {
+    using namespace DGtal;
+
+    // trace.beginBlock ( "Create Fixture_object_diamond" );
+    Point p1( -10, -10, -10 );
+    Point p2( 10, 10, 10 );
+    Domain domain( p1, p2 );
+    Point c( 0, 0, 0 );
+
+    // diamond of radius 4
+    FixtureDigitalSet diamond_set( domain );
+    for ( auto it = domain.begin(); it != domain.end(); ++it )
+    {
+      if ( (*it - c ).norm1() <= 3 ) diamond_set.insertNew( *it );
+    }
+    diamond_set.erase( c );
+
+    KSpace K;
```

Comment:
I think you do not use cells in your test file. I suggest to remove all lines with KSpace, SurfelAdjacency, etc.